{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "os.chdir(\"/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/\")\n",
    "os.getcwd()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Algae studies of interest\n",
    "### Red Algae\n",
    "MGYS00003330: \n",
    "- 1 Samples\n",
    "- EMG produced TPA metagenomics assembly of the Marine algal microbial communities from Sidmouth, United Kingdom - Sidmouth_Asex2 metaG metagenome (marine metagenome) data set.\n",
    "\n",
    "MGYS00003317: \n",
    "- 1 Samples\n",
    "- EMG produced TPA metagenomics assembly of the Marine algal microbial communities from Porto, Italy - Porto_4 metaG metagenome (marine metagenome) data set.\n",
    "\n",
    "### Brown Algae\n",
    "MGYS00002998: \n",
    "- 22 samples\n",
    "- Description: The ecology of microbial communities associated with Macrosystis pyrifera\n",
    "- Publication: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0067480\n",
    "- Sequences were first screened for quality using the following parameters: minimum quality score of 25, minimum sequence length of 200 bp, maximum length of 1000 bp, and no ambiguous bases in the entire sequence or mismatches in the primer sequence. Any sequences not meeting these parameters were excluded from downstream analyses. Sequences were then sorted by barcode into their respective samples and the barcode and primer sequences were removed. The sequences were denoised using the QIIME denoiser [17] and operational taxonomic unites (OTUs) were clustered de novo from the denoised sequences using uclust [18] at 97% identity.\n",
    "- Taxonomy was assigned using BLAST against the Silva database (prefiltered at 97% identity). Chimeras were removed from the reference set on the basis of identification as chimeric via ChimeraSlayer [20].\n",
    "\n",
    "MGYS00002643:\n",
    "- 96 samples\n",
    "- Biological Impacts of Ocean Acidification\n",
    "\n",
    "MGYS00001287:\n",
    "- 12 Samples\n",
    "- scs diatom\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import os\n",
    "from src.extract import fetch_metadata, fetch_study_data, get_sample_analyses, handle_analysis_summary, check_available_sample_analysis_data, get_data_file, get_SSU_tsv_file_list_for_study, get_tsv_files\n",
    "from src.plot import add_data_to_map\n",
    "from src.transform import read_otu_tsv_file, build_otu_lookup_df, build_otu_count_df_and_lookup_df\n",
    "from src.todo import join_on_sample_id_sub_select_df_columns, set_row_color_by_term_presence\n",
    "study_id = \"MGYS00002998\"\n",
    "endpoint = f\"studies/{study_id}\"\n",
    "api_root = \"https://www.ebi.ac.uk/metagenomics/api/v1/\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get Study Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>id</th>\n",
       "      <th>attributes.samples-count</th>\n",
       "      <th>attributes.bioproject</th>\n",
       "      <th>attributes.accession</th>\n",
       "      <th>attributes.is-private</th>\n",
       "      <th>attributes.secondary-accession</th>\n",
       "      <th>attributes.centre-name</th>\n",
       "      <th>attributes.study-abstract</th>\n",
       "      <th>attributes.study-name</th>\n",
       "      <th>attributes.data-origination</th>\n",
       "      <th>attributes.last-update</th>\n",
       "      <th>relationships.biomes.data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>studies</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>22</td>\n",
       "      <td>PRJEB1154</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>False</td>\n",
       "      <td>ERP002019</td>\n",
       "      <td>CCME-COLORADO</td>\n",
       "      <td>Bacteria control major nutrient cycles and dir...</td>\n",
       "      <td>The ecology of microbial communities associate...</td>\n",
       "      <td>SUBMITTED</td>\n",
       "      <td>2018-09-12T11:19:58</td>\n",
       "      <td>[{'id': 'root:Host-associated:Algae:Brown Alga...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      type            id  attributes.samples-count attributes.bioproject  \\\n",
       "0  studies  MGYS00002998                        22             PRJEB1154   \n",
       "\n",
       "  attributes.accession  attributes.is-private attributes.secondary-accession  \\\n",
       "0         MGYS00002998                  False                      ERP002019   \n",
       "\n",
       "  attributes.centre-name                          attributes.study-abstract  \\\n",
       "0          CCME-COLORADO  Bacteria control major nutrient cycles and dir...   \n",
       "\n",
       "                               attributes.study-name  \\\n",
       "0  The ecology of microbial communities associate...   \n",
       "\n",
       "  attributes.data-origination attributes.last-update  \\\n",
       "0                   SUBMITTED    2018-09-12T11:19:58   \n",
       "\n",
       "                           relationships.biomes.data  \n",
       "0  [{'id': 'root:Host-associated:Algae:Brown Alga...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "study_metadata = fetch_metadata(endpoint, api_root)\n",
    "study_metadata\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get samples associated with a study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fetching MGYS00002998 samples\n",
      "       type         id  attributes.longitude  attributes.latitude  \\\n",
      "0   samples  ERS196395                -121.9              36.6167   \n",
      "1   samples  ERS196396                -121.9              36.6167   \n",
      "2   samples  ERS196397                -121.9              36.6167   \n",
      "3   samples  ERS196398                -121.9              36.6167   \n",
      "4   samples  ERS196399                -121.9              36.6167   \n",
      "5   samples  ERS196400                -121.9              36.6167   \n",
      "6   samples  ERS196401                -121.9              36.6167   \n",
      "7   samples  ERS196402                -121.9              36.6167   \n",
      "8   samples  ERS196403                -121.9              36.6167   \n",
      "9   samples  ERS196404                -121.9              36.6167   \n",
      "10  samples  ERS196405                -121.9              36.6167   \n",
      "11  samples  ERS196406                -121.9              36.6167   \n",
      "12  samples  ERS196407                -121.9              36.6167   \n",
      "13  samples  ERS196408                -121.9              36.6167   \n",
      "14  samples  ERS196409                -121.9              36.6167   \n",
      "15  samples  ERS196410                -121.9              36.6167   \n",
      "16  samples  ERS196411                -121.9              36.6167   \n",
      "17  samples  ERS196412                -121.9              36.6167   \n",
      "18  samples  ERS196413                -121.9              36.6167   \n",
      "19  samples  ERS196414                -121.9              36.6167   \n",
      "20  samples  ERS196415                -121.9              36.6167   \n",
      "21  samples  ERS196416                -121.9              36.6167   \n",
      "\n",
      "   attributes.biosample                         attributes.sample-metadata  \\\n",
      "0          SAMEA1696088  [{'key': 'geographic location (longitude)', 'v...   \n",
      "1          SAMEA1696082  [{'key': 'geographic location (longitude)', 'v...   \n",
      "2          SAMEA1696095  [{'key': 'geographic location (longitude)', 'v...   \n",
      "3          SAMEA1696090  [{'key': 'geographic location (longitude)', 'v...   \n",
      "4          SAMEA1696086  [{'key': 'geographic location (longitude)', 'v...   \n",
      "5          SAMEA1696083  [{'key': 'geographic location (longitude)', 'v...   \n",
      "6          SAMEA1696080  [{'key': 'geographic location (longitude)', 'v...   \n",
      "7          SAMEA1696076  [{'key': 'geographic location (longitude)', 'v...   \n",
      "8          SAMEA1696089  [{'key': 'geographic location (longitude)', 'v...   \n",
      "9          SAMEA1696093  [{'key': 'geographic location (longitude)', 'v...   \n",
      "10         SAMEA1696091  [{'key': 'geographic location (longitude)', 'v...   \n",
      "11         SAMEA1696081  [{'key': 'geographic location (longitude)', 'v...   \n",
      "12         SAMEA1696077  [{'key': 'geographic location (longitude)', 'v...   \n",
      "13         SAMEA1696094  [{'key': 'geographic location (longitude)', 'v...   \n",
      "14         SAMEA1696078  [{'key': 'geographic location (longitude)', 'v...   \n",
      "15         SAMEA1696079  [{'key': 'geographic location (longitude)', 'v...   \n",
      "16         SAMEA1696097  [{'key': 'geographic location (longitude)', 'v...   \n",
      "17         SAMEA1696085  [{'key': 'geographic location (longitude)', 'v...   \n",
      "18         SAMEA1696087  [{'key': 'geographic location (longitude)', 'v...   \n",
      "19         SAMEA1696084  [{'key': 'geographic location (longitude)', 'v...   \n",
      "20         SAMEA1696092  [{'key': 'geographic location (longitude)', 'v...   \n",
      "21         SAMEA1696096  [{'key': 'geographic location (longitude)', 'v...   \n",
      "\n",
      "   attributes.accession attributes.analysis-completed  \\\n",
      "0             ERS196395                    2018-09-12   \n",
      "1             ERS196396                    2018-09-12   \n",
      "2             ERS196397                    2018-09-12   \n",
      "3             ERS196398                    2018-09-12   \n",
      "4             ERS196399                    2018-09-12   \n",
      "5             ERS196400                    2018-09-12   \n",
      "6             ERS196401                    2018-09-12   \n",
      "7             ERS196402                    2018-09-12   \n",
      "8             ERS196403                    2018-09-12   \n",
      "9             ERS196404                    2018-09-12   \n",
      "10            ERS196405                    2018-09-12   \n",
      "11            ERS196406                    2018-09-12   \n",
      "12            ERS196407                    2018-09-12   \n",
      "13            ERS196408                    2018-09-12   \n",
      "14            ERS196409                    2018-09-12   \n",
      "15            ERS196410                    2018-09-12   \n",
      "16            ERS196411                    2018-09-12   \n",
      "17            ERS196412                    2018-09-12   \n",
      "18            ERS196413                    2018-09-12   \n",
      "19            ERS196414                    2018-09-12   \n",
      "20            ERS196415                    2018-09-12   \n",
      "21            ERS196416                    2018-09-12   \n",
      "\n",
      "         attributes.geo-loc-name  \\\n",
      "0   GAZ:United States of America   \n",
      "1   GAZ:United States of America   \n",
      "2   GAZ:United States of America   \n",
      "3   GAZ:United States of America   \n",
      "4   GAZ:United States of America   \n",
      "5   GAZ:United States of America   \n",
      "6   GAZ:United States of America   \n",
      "7   GAZ:United States of America   \n",
      "8   GAZ:United States of America   \n",
      "9   GAZ:United States of America   \n",
      "10  GAZ:United States of America   \n",
      "11  GAZ:United States of America   \n",
      "12  GAZ:United States of America   \n",
      "13  GAZ:United States of America   \n",
      "14  GAZ:United States of America   \n",
      "15  GAZ:United States of America   \n",
      "16  GAZ:United States of America   \n",
      "17  GAZ:United States of America   \n",
      "18  GAZ:United States of America   \n",
      "19  GAZ:United States of America   \n",
      "20  GAZ:United States of America   \n",
      "21  GAZ:United States of America   \n",
      "\n",
      "                               attributes.sample-desc  \\\n",
      "0                   Bacterial communities in seawater   \n",
      "1                   Bacterial communities in seawater   \n",
      "2                   Bacterial communities in seawater   \n",
      "3   Bacterial biofilms on the surface of Macrocyst...   \n",
      "4                   Bacterial communities in seawater   \n",
      "5                   Bacterial communities in seawater   \n",
      "6   Bacterial biofilms on the surface of Macrocyst...   \n",
      "7                   Bacterial communities in seawater   \n",
      "8   Bacterial biofilms on the surface of Macrocyst...   \n",
      "9                   Bacterial communities in seawater   \n",
      "10                  Bacterial communities in seawater   \n",
      "11  Bacterial biofilms on the surface of Macrocyst...   \n",
      "12                  Bacterial communities in seawater   \n",
      "13                  Bacterial communities in seawater   \n",
      "14                  Bacterial communities in seawater   \n",
      "15                  Bacterial communities in seawater   \n",
      "16  Bacterial biofilms on the surface of Macrocyst...   \n",
      "17                  Bacterial communities in seawater   \n",
      "18                  Bacterial communities in seawater   \n",
      "19  Bacterial biofilms on the surface of Macrocyst...   \n",
      "20                  Bacterial communities in seawater   \n",
      "21                  Bacterial communities in seawater   \n",
      "\n",
      "   attributes.environment-biome attributes.environment-feature  \\\n",
      "0             ENVO:marine biome             ENVO:coastal water   \n",
      "1              ENVO:ocean water             ENVO:coastal water   \n",
      "2             ENVO:marine biome             ENVO:coastal water   \n",
      "3             ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "4             ENVO:marine biome             ENVO:coastal water   \n",
      "5              ENVO:ocean water             ENVO:coastal water   \n",
      "6             ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "7              ENVO:ocean water             ENVO:coastal water   \n",
      "8             ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "9              ENVO:ocean water             ENVO:coastal water   \n",
      "10            ENVO:marine biome             ENVO:coastal water   \n",
      "11            ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "12             ENVO:ocean water             ENVO:coastal water   \n",
      "13            ENVO:marine biome             ENVO:coastal water   \n",
      "14             ENVO:ocean water             ENVO:coastal water   \n",
      "15            ENVO:marine biome             ENVO:coastal water   \n",
      "16            ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "17            ENVO:marine biome             ENVO:coastal water   \n",
      "18             ENVO:ocean water             ENVO:coastal water   \n",
      "19            ENVO:marine biome  ENVO:plant-associated habitat   \n",
      "20            ENVO:marine biome             ENVO:coastal water   \n",
      "21            ENVO:marine biome             ENVO:coastal water   \n",
      "\n",
      "   attributes.environment-material attributes.sample-name  \\\n",
      "0                       ENVO:water              01.2water   \n",
      "1                       ENVO:water              11.1water   \n",
      "2                       ENVO:water              07.1water   \n",
      "3                     ENVO:biofilm              09.2macro   \n",
      "4                       ENVO:water              05.2water   \n",
      "5                       ENVO:water              08.1water   \n",
      "6                     ENVO:biofilm              03.2macro   \n",
      "7                       ENVO:water              10.2water   \n",
      "8                     ENVO:biofilm              12.2macro   \n",
      "9                       ENVO:water              07.2water   \n",
      "10                      ENVO:water              02.1water   \n",
      "11                    ENVO:biofilm              12.1macro   \n",
      "12                      ENVO:water              11.2water   \n",
      "13                      ENVO:water              04.2water   \n",
      "14                      ENVO:water              10.1water   \n",
      "15                      ENVO:water              02.2water   \n",
      "16                    ENVO:biofilm              03.1macro   \n",
      "17                      ENVO:water              01.1water   \n",
      "18                      ENVO:water              05.1water   \n",
      "19                    ENVO:biofilm              09.1macro   \n",
      "20                      ENVO:water              08.2water   \n",
      "21                      ENVO:water              04.1water   \n",
      "\n",
      "      attributes.sample-alias attributes.last-update  \\\n",
      "0   qiime_study_820:01.2water    2018-09-12T11:20:06   \n",
      "1   qiime_study_820:11.1water    2018-09-12T11:20:06   \n",
      "2   qiime_study_820:07.1water    2018-09-12T11:20:06   \n",
      "3   qiime_study_820:09.2macro    2018-09-12T11:20:06   \n",
      "4   qiime_study_820:05.2water    2018-09-12T11:20:06   \n",
      "5   qiime_study_820:08.1water    2018-09-12T11:20:06   \n",
      "6   qiime_study_820:03.2macro    2018-09-12T11:20:06   \n",
      "7   qiime_study_820:10.2water    2018-09-12T11:20:06   \n",
      "8   qiime_study_820:12.2macro    2018-09-12T11:20:06   \n",
      "9   qiime_study_820:07.2water    2018-09-12T11:20:06   \n",
      "10  qiime_study_820:02.1water    2018-09-12T11:20:06   \n",
      "11  qiime_study_820:12.1macro    2018-09-12T11:20:06   \n",
      "12  qiime_study_820:11.2water    2018-09-12T11:20:06   \n",
      "13  qiime_study_820:04.2water    2018-09-12T11:20:06   \n",
      "14  qiime_study_820:10.1water    2018-09-12T11:20:06   \n",
      "15  qiime_study_820:02.2water    2018-09-12T11:20:06   \n",
      "16  qiime_study_820:03.1macro    2018-09-12T11:20:06   \n",
      "17  qiime_study_820:01.1water    2018-09-12T11:20:06   \n",
      "18  qiime_study_820:05.1water    2018-09-12T11:20:06   \n",
      "19  qiime_study_820:09.1macro    2018-09-12T11:20:06   \n",
      "20  qiime_study_820:08.2water    2018-09-12T11:20:06   \n",
      "21  qiime_study_820:04.1water    2018-09-12T11:20:06   \n",
      "\n",
      "                     relationships.studies.data  \\\n",
      "0   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "1   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "2   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "3   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "4   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "5   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "6   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "7   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "8   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "9   [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "10  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "11  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "12  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "13  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "14  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "15  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "16  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "17  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "18  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "19  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "20  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "21  [{'id': 'MGYS00002998', 'type': 'studies'}]   \n",
      "\n",
      "               relationships.biome.data.id relationships.biome.data.type  \n",
      "0   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "1   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "2   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "3   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "4   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "5   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "6   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "7   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "8   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "9   root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "10  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "11  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "12  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "13  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "14  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "15  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "16  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "17  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "18  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "19  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "20  root:Host-associated:Algae:Brown Algae                        biomes  \n",
      "21  root:Host-associated:Algae:Brown Algae                        biomes  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>study</th>\n",
       "      <th>lon</th>\n",
       "      <th>lat</th>\n",
       "      <th>sample_description</th>\n",
       "      <th>biome</th>\n",
       "      <th>environment</th>\n",
       "      <th>biome_relationship</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ERS196395</th>\n",
       "      <td>ERS196395</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196396</th>\n",
       "      <td>ERS196396</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196397</th>\n",
       "      <td>ERS196397</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196398</th>\n",
       "      <td>ERS196398</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196399</th>\n",
       "      <td>ERS196399</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196400</th>\n",
       "      <td>ERS196400</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196401</th>\n",
       "      <td>ERS196401</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196402</th>\n",
       "      <td>ERS196402</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196403</th>\n",
       "      <td>ERS196403</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196404</th>\n",
       "      <td>ERS196404</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196405</th>\n",
       "      <td>ERS196405</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196406</th>\n",
       "      <td>ERS196406</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196407</th>\n",
       "      <td>ERS196407</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196408</th>\n",
       "      <td>ERS196408</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196409</th>\n",
       "      <td>ERS196409</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196410</th>\n",
       "      <td>ERS196410</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196411</th>\n",
       "      <td>ERS196411</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196412</th>\n",
       "      <td>ERS196412</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196413</th>\n",
       "      <td>ERS196413</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:ocean water</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196414</th>\n",
       "      <td>ERS196414</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial biofilms on the surface of Macrocyst...</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:plant-associated habitat</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196415</th>\n",
       "      <td>ERS196415</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERS196416</th>\n",
       "      <td>ERS196416</td>\n",
       "      <td>MGYS00002998</td>\n",
       "      <td>-121.9</td>\n",
       "      <td>36.6167</td>\n",
       "      <td>Bacterial communities in seawater</td>\n",
       "      <td>ENVO:marine biome</td>\n",
       "      <td>ENVO:coastal water</td>\n",
       "      <td>root:Host-associated:Algae:Brown Algae</td>\n",
       "      <td>#FF0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           sample_id         study    lon      lat  \\\n",
       "accession                                            \n",
       "ERS196395  ERS196395  MGYS00002998 -121.9  36.6167   \n",
       "ERS196396  ERS196396  MGYS00002998 -121.9  36.6167   \n",
       "ERS196397  ERS196397  MGYS00002998 -121.9  36.6167   \n",
       "ERS196398  ERS196398  MGYS00002998 -121.9  36.6167   \n",
       "ERS196399  ERS196399  MGYS00002998 -121.9  36.6167   \n",
       "ERS196400  ERS196400  MGYS00002998 -121.9  36.6167   \n",
       "ERS196401  ERS196401  MGYS00002998 -121.9  36.6167   \n",
       "ERS196402  ERS196402  MGYS00002998 -121.9  36.6167   \n",
       "ERS196403  ERS196403  MGYS00002998 -121.9  36.6167   \n",
       "ERS196404  ERS196404  MGYS00002998 -121.9  36.6167   \n",
       "ERS196405  ERS196405  MGYS00002998 -121.9  36.6167   \n",
       "ERS196406  ERS196406  MGYS00002998 -121.9  36.6167   \n",
       "ERS196407  ERS196407  MGYS00002998 -121.9  36.6167   \n",
       "ERS196408  ERS196408  MGYS00002998 -121.9  36.6167   \n",
       "ERS196409  ERS196409  MGYS00002998 -121.9  36.6167   \n",
       "ERS196410  ERS196410  MGYS00002998 -121.9  36.6167   \n",
       "ERS196411  ERS196411  MGYS00002998 -121.9  36.6167   \n",
       "ERS196412  ERS196412  MGYS00002998 -121.9  36.6167   \n",
       "ERS196413  ERS196413  MGYS00002998 -121.9  36.6167   \n",
       "ERS196414  ERS196414  MGYS00002998 -121.9  36.6167   \n",
       "ERS196415  ERS196415  MGYS00002998 -121.9  36.6167   \n",
       "ERS196416  ERS196416  MGYS00002998 -121.9  36.6167   \n",
       "\n",
       "                                          sample_description  \\\n",
       "accession                                                      \n",
       "ERS196395                  Bacterial communities in seawater   \n",
       "ERS196396                  Bacterial communities in seawater   \n",
       "ERS196397                  Bacterial communities in seawater   \n",
       "ERS196398  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196399                  Bacterial communities in seawater   \n",
       "ERS196400                  Bacterial communities in seawater   \n",
       "ERS196401  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196402                  Bacterial communities in seawater   \n",
       "ERS196403  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196404                  Bacterial communities in seawater   \n",
       "ERS196405                  Bacterial communities in seawater   \n",
       "ERS196406  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196407                  Bacterial communities in seawater   \n",
       "ERS196408                  Bacterial communities in seawater   \n",
       "ERS196409                  Bacterial communities in seawater   \n",
       "ERS196410                  Bacterial communities in seawater   \n",
       "ERS196411  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196412                  Bacterial communities in seawater   \n",
       "ERS196413                  Bacterial communities in seawater   \n",
       "ERS196414  Bacterial biofilms on the surface of Macrocyst...   \n",
       "ERS196415                  Bacterial communities in seawater   \n",
       "ERS196416                  Bacterial communities in seawater   \n",
       "\n",
       "                       biome                    environment  \\\n",
       "accession                                                     \n",
       "ERS196395  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196396   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196397  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196398  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196399  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196400   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196401  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196402   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196403  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196404   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196405  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196406  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196407   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196408  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196409   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196410  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196411  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196412  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196413   ENVO:ocean water             ENVO:coastal water   \n",
       "ERS196414  ENVO:marine biome  ENVO:plant-associated habitat   \n",
       "ERS196415  ENVO:marine biome             ENVO:coastal water   \n",
       "ERS196416  ENVO:marine biome             ENVO:coastal water   \n",
       "\n",
       "                               biome_relationship    color  \n",
       "accession                                                   \n",
       "ERS196395  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196396  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196397  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196398  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196399  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196400  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196401  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196402  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196403  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196404  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196405  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196406  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196407  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196408  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196409  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196410  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196411  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196412  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196413  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196414  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196415  root:Host-associated:Algae:Brown Algae  #FF0000  \n",
       "ERS196416  root:Host-associated:Algae:Brown Algae  #FF0000  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_samples = fetch_study_data(api_root, study_metadata, sample_count=100)\n",
    "study_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get analyses associated with study samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THERE ARE 22 being added to the df\n",
      "the df is (22, 20)\n",
      "but now the df is (22, 30)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "22"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analyses = get_sample_analyses(api_root, study_samples, analysis_limit=100)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>id</th>\n",
       "      <th>attributes.accession</th>\n",
       "      <th>attributes.analysis-status</th>\n",
       "      <th>attributes.analysis-summary</th>\n",
       "      <th>attributes.experiment-type</th>\n",
       "      <th>attributes.pipeline-version</th>\n",
       "      <th>attributes.is-private</th>\n",
       "      <th>attributes.complete-time</th>\n",
       "      <th>attributes.instrument-platform</th>\n",
       "      <th>...</th>\n",
       "      <th>Submitted nucleotide sequences</th>\n",
       "      <th>Nucleotide sequences after format-specific filtering</th>\n",
       "      <th>Nucleotide sequences after length filtering</th>\n",
       "      <th>Nucleotide sequences after undetermined bases filtering</th>\n",
       "      <th>Reads with predicted CDS</th>\n",
       "      <th>Reads with predicted RNA</th>\n",
       "      <th>Reads with InterProScan match</th>\n",
       "      <th>Predicted CDS</th>\n",
       "      <th>Predicted CDS with InterProScan match</th>\n",
       "      <th>Total InterProScan matches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212985</td>\n",
       "      <td>MGYA00212985</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>19112</td>\n",
       "      <td>19111</td>\n",
       "      <td>17969</td>\n",
       "      <td>17969</td>\n",
       "      <td>595</td>\n",
       "      <td>17255</td>\n",
       "      <td>0</td>\n",
       "      <td>598</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212971</td>\n",
       "      <td>MGYA00212971</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>23787</td>\n",
       "      <td>23785</td>\n",
       "      <td>22474</td>\n",
       "      <td>22474</td>\n",
       "      <td>966</td>\n",
       "      <td>21302</td>\n",
       "      <td>0</td>\n",
       "      <td>967</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212970</td>\n",
       "      <td>MGYA00212970</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>19767</td>\n",
       "      <td>19765</td>\n",
       "      <td>18279</td>\n",
       "      <td>18279</td>\n",
       "      <td>861</td>\n",
       "      <td>17239</td>\n",
       "      <td>0</td>\n",
       "      <td>864</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212976</td>\n",
       "      <td>MGYA00212976</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>6501</td>\n",
       "      <td>6501</td>\n",
       "      <td>5510</td>\n",
       "      <td>5510</td>\n",
       "      <td>328</td>\n",
       "      <td>5020</td>\n",
       "      <td>0</td>\n",
       "      <td>328</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212979</td>\n",
       "      <td>MGYA00212979</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>21099</td>\n",
       "      <td>21099</td>\n",
       "      <td>20172</td>\n",
       "      <td>20172</td>\n",
       "      <td>737</td>\n",
       "      <td>19281</td>\n",
       "      <td>0</td>\n",
       "      <td>738</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212974</td>\n",
       "      <td>MGYA00212974</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>23829</td>\n",
       "      <td>23828</td>\n",
       "      <td>22676</td>\n",
       "      <td>22676</td>\n",
       "      <td>1059</td>\n",
       "      <td>21437</td>\n",
       "      <td>0</td>\n",
       "      <td>1061</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212980</td>\n",
       "      <td>MGYA00212980</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>40580</td>\n",
       "      <td>40580</td>\n",
       "      <td>36470</td>\n",
       "      <td>36470</td>\n",
       "      <td>4175</td>\n",
       "      <td>30993</td>\n",
       "      <td>0</td>\n",
       "      <td>4179</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212988</td>\n",
       "      <td>MGYA00212988</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>25517</td>\n",
       "      <td>25516</td>\n",
       "      <td>24135</td>\n",
       "      <td>24135</td>\n",
       "      <td>1176</td>\n",
       "      <td>22717</td>\n",
       "      <td>0</td>\n",
       "      <td>1176</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212972</td>\n",
       "      <td>MGYA00212972</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>3686</td>\n",
       "      <td>3686</td>\n",
       "      <td>3262</td>\n",
       "      <td>3262</td>\n",
       "      <td>327</td>\n",
       "      <td>2637</td>\n",
       "      <td>0</td>\n",
       "      <td>327</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212983</td>\n",
       "      <td>MGYA00212983</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>27486</td>\n",
       "      <td>27485</td>\n",
       "      <td>25561</td>\n",
       "      <td>25561</td>\n",
       "      <td>1308</td>\n",
       "      <td>24007</td>\n",
       "      <td>0</td>\n",
       "      <td>1310</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212984</td>\n",
       "      <td>MGYA00212984</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>31018</td>\n",
       "      <td>31018</td>\n",
       "      <td>28985</td>\n",
       "      <td>28985</td>\n",
       "      <td>1089</td>\n",
       "      <td>27744</td>\n",
       "      <td>0</td>\n",
       "      <td>1092</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212989</td>\n",
       "      <td>MGYA00212989</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>7167</td>\n",
       "      <td>7167</td>\n",
       "      <td>6196</td>\n",
       "      <td>6196</td>\n",
       "      <td>568</td>\n",
       "      <td>5392</td>\n",
       "      <td>0</td>\n",
       "      <td>568</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212973</td>\n",
       "      <td>MGYA00212973</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>24559</td>\n",
       "      <td>24559</td>\n",
       "      <td>23159</td>\n",
       "      <td>23159</td>\n",
       "      <td>1071</td>\n",
       "      <td>21901</td>\n",
       "      <td>1</td>\n",
       "      <td>1071</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212986</td>\n",
       "      <td>MGYA00212986</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>27926</td>\n",
       "      <td>27926</td>\n",
       "      <td>26443</td>\n",
       "      <td>26443</td>\n",
       "      <td>1052</td>\n",
       "      <td>25164</td>\n",
       "      <td>0</td>\n",
       "      <td>1054</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212975</td>\n",
       "      <td>MGYA00212975</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>23666</td>\n",
       "      <td>23665</td>\n",
       "      <td>21988</td>\n",
       "      <td>21988</td>\n",
       "      <td>1015</td>\n",
       "      <td>20800</td>\n",
       "      <td>3</td>\n",
       "      <td>1015</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212977</td>\n",
       "      <td>MGYA00212977</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>26000</td>\n",
       "      <td>26000</td>\n",
       "      <td>24423</td>\n",
       "      <td>24423</td>\n",
       "      <td>888</td>\n",
       "      <td>23372</td>\n",
       "      <td>0</td>\n",
       "      <td>888</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212982</td>\n",
       "      <td>MGYA00212982</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>16135</td>\n",
       "      <td>16135</td>\n",
       "      <td>14415</td>\n",
       "      <td>14415</td>\n",
       "      <td>1179</td>\n",
       "      <td>12960</td>\n",
       "      <td>0</td>\n",
       "      <td>1180</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212981</td>\n",
       "      <td>MGYA00212981</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>26225</td>\n",
       "      <td>26225</td>\n",
       "      <td>24449</td>\n",
       "      <td>24449</td>\n",
       "      <td>823</td>\n",
       "      <td>23460</td>\n",
       "      <td>0</td>\n",
       "      <td>823</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212987</td>\n",
       "      <td>MGYA00212987</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>6856</td>\n",
       "      <td>6856</td>\n",
       "      <td>6595</td>\n",
       "      <td>6595</td>\n",
       "      <td>193</td>\n",
       "      <td>6347</td>\n",
       "      <td>0</td>\n",
       "      <td>193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212990</td>\n",
       "      <td>MGYA00212990</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>22018</td>\n",
       "      <td>22017</td>\n",
       "      <td>19356</td>\n",
       "      <td>19356</td>\n",
       "      <td>1399</td>\n",
       "      <td>17215</td>\n",
       "      <td>0</td>\n",
       "      <td>1400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212991</td>\n",
       "      <td>MGYA00212991</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>22276</td>\n",
       "      <td>22275</td>\n",
       "      <td>21072</td>\n",
       "      <td>21072</td>\n",
       "      <td>941</td>\n",
       "      <td>19951</td>\n",
       "      <td>0</td>\n",
       "      <td>942</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>analysis-jobs</td>\n",
       "      <td>MGYA00212978</td>\n",
       "      <td>MGYA00212978</td>\n",
       "      <td>completed</td>\n",
       "      <td>[{'key': 'Submitted nucleotide sequences', 'va...</td>\n",
       "      <td>amplicon</td>\n",
       "      <td>4.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-09-12T00:00:00</td>\n",
       "      <td>LS454</td>\n",
       "      <td>...</td>\n",
       "      <td>27473</td>\n",
       "      <td>27472</td>\n",
       "      <td>26044</td>\n",
       "      <td>26044</td>\n",
       "      <td>952</td>\n",
       "      <td>24892</td>\n",
       "      <td>0</td>\n",
       "      <td>954</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            type            id attributes.accession  \\\n",
       "0  analysis-jobs  MGYA00212985         MGYA00212985   \n",
       "0  analysis-jobs  MGYA00212971         MGYA00212971   \n",
       "0  analysis-jobs  MGYA00212970         MGYA00212970   \n",
       "0  analysis-jobs  MGYA00212976         MGYA00212976   \n",
       "0  analysis-jobs  MGYA00212979         MGYA00212979   \n",
       "0  analysis-jobs  MGYA00212974         MGYA00212974   \n",
       "0  analysis-jobs  MGYA00212980         MGYA00212980   \n",
       "0  analysis-jobs  MGYA00212988         MGYA00212988   \n",
       "0  analysis-jobs  MGYA00212972         MGYA00212972   \n",
       "0  analysis-jobs  MGYA00212983         MGYA00212983   \n",
       "0  analysis-jobs  MGYA00212984         MGYA00212984   \n",
       "0  analysis-jobs  MGYA00212989         MGYA00212989   \n",
       "0  analysis-jobs  MGYA00212973         MGYA00212973   \n",
       "0  analysis-jobs  MGYA00212986         MGYA00212986   \n",
       "0  analysis-jobs  MGYA00212975         MGYA00212975   \n",
       "0  analysis-jobs  MGYA00212977         MGYA00212977   \n",
       "0  analysis-jobs  MGYA00212982         MGYA00212982   \n",
       "0  analysis-jobs  MGYA00212981         MGYA00212981   \n",
       "0  analysis-jobs  MGYA00212987         MGYA00212987   \n",
       "0  analysis-jobs  MGYA00212990         MGYA00212990   \n",
       "0  analysis-jobs  MGYA00212991         MGYA00212991   \n",
       "0  analysis-jobs  MGYA00212978         MGYA00212978   \n",
       "\n",
       "  attributes.analysis-status  \\\n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "0                  completed   \n",
       "\n",
       "                         attributes.analysis-summary  \\\n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "0  [{'key': 'Submitted nucleotide sequences', 'va...   \n",
       "\n",
       "  attributes.experiment-type attributes.pipeline-version  \\\n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "0                   amplicon                         4.1   \n",
       "\n",
       "   attributes.is-private attributes.complete-time  \\\n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "0                  False      2018-09-12T00:00:00   \n",
       "\n",
       "  attributes.instrument-platform  ... Submitted nucleotide sequences  \\\n",
       "0                          LS454  ...                          19112   \n",
       "0                          LS454  ...                          23787   \n",
       "0                          LS454  ...                          19767   \n",
       "0                          LS454  ...                           6501   \n",
       "0                          LS454  ...                          21099   \n",
       "0                          LS454  ...                          23829   \n",
       "0                          LS454  ...                          40580   \n",
       "0                          LS454  ...                          25517   \n",
       "0                          LS454  ...                           3686   \n",
       "0                          LS454  ...                          27486   \n",
       "0                          LS454  ...                          31018   \n",
       "0                          LS454  ...                           7167   \n",
       "0                          LS454  ...                          24559   \n",
       "0                          LS454  ...                          27926   \n",
       "0                          LS454  ...                          23666   \n",
       "0                          LS454  ...                          26000   \n",
       "0                          LS454  ...                          16135   \n",
       "0                          LS454  ...                          26225   \n",
       "0                          LS454  ...                           6856   \n",
       "0                          LS454  ...                          22018   \n",
       "0                          LS454  ...                          22276   \n",
       "0                          LS454  ...                          27473   \n",
       "\n",
       "  Nucleotide sequences after format-specific filtering  \\\n",
       "0                                              19111     \n",
       "0                                              23785     \n",
       "0                                              19765     \n",
       "0                                               6501     \n",
       "0                                              21099     \n",
       "0                                              23828     \n",
       "0                                              40580     \n",
       "0                                              25516     \n",
       "0                                               3686     \n",
       "0                                              27485     \n",
       "0                                              31018     \n",
       "0                                               7167     \n",
       "0                                              24559     \n",
       "0                                              27926     \n",
       "0                                              23665     \n",
       "0                                              26000     \n",
       "0                                              16135     \n",
       "0                                              26225     \n",
       "0                                               6856     \n",
       "0                                              22017     \n",
       "0                                              22275     \n",
       "0                                              27472     \n",
       "\n",
       "  Nucleotide sequences after length filtering  \\\n",
       "0                                       17969   \n",
       "0                                       22474   \n",
       "0                                       18279   \n",
       "0                                        5510   \n",
       "0                                       20172   \n",
       "0                                       22676   \n",
       "0                                       36470   \n",
       "0                                       24135   \n",
       "0                                        3262   \n",
       "0                                       25561   \n",
       "0                                       28985   \n",
       "0                                        6196   \n",
       "0                                       23159   \n",
       "0                                       26443   \n",
       "0                                       21988   \n",
       "0                                       24423   \n",
       "0                                       14415   \n",
       "0                                       24449   \n",
       "0                                        6595   \n",
       "0                                       19356   \n",
       "0                                       21072   \n",
       "0                                       26044   \n",
       "\n",
       "  Nucleotide sequences after undetermined bases filtering  \\\n",
       "0                                              17969        \n",
       "0                                              22474        \n",
       "0                                              18279        \n",
       "0                                               5510        \n",
       "0                                              20172        \n",
       "0                                              22676        \n",
       "0                                              36470        \n",
       "0                                              24135        \n",
       "0                                               3262        \n",
       "0                                              25561        \n",
       "0                                              28985        \n",
       "0                                               6196        \n",
       "0                                              23159        \n",
       "0                                              26443        \n",
       "0                                              21988        \n",
       "0                                              24423        \n",
       "0                                              14415        \n",
       "0                                              24449        \n",
       "0                                               6595        \n",
       "0                                              19356        \n",
       "0                                              21072        \n",
       "0                                              26044        \n",
       "\n",
       "  Reads with predicted CDS Reads with predicted RNA  \\\n",
       "0                      595                    17255   \n",
       "0                      966                    21302   \n",
       "0                      861                    17239   \n",
       "0                      328                     5020   \n",
       "0                      737                    19281   \n",
       "0                     1059                    21437   \n",
       "0                     4175                    30993   \n",
       "0                     1176                    22717   \n",
       "0                      327                     2637   \n",
       "0                     1308                    24007   \n",
       "0                     1089                    27744   \n",
       "0                      568                     5392   \n",
       "0                     1071                    21901   \n",
       "0                     1052                    25164   \n",
       "0                     1015                    20800   \n",
       "0                      888                    23372   \n",
       "0                     1179                    12960   \n",
       "0                      823                    23460   \n",
       "0                      193                     6347   \n",
       "0                     1399                    17215   \n",
       "0                      941                    19951   \n",
       "0                      952                    24892   \n",
       "\n",
       "  Reads with InterProScan match Predicted CDS  \\\n",
       "0                             0           598   \n",
       "0                             0           967   \n",
       "0                             0           864   \n",
       "0                             0           328   \n",
       "0                             0           738   \n",
       "0                             0          1061   \n",
       "0                             0          4179   \n",
       "0                             0          1176   \n",
       "0                             0           327   \n",
       "0                             0          1310   \n",
       "0                             0          1092   \n",
       "0                             0           568   \n",
       "0                             1          1071   \n",
       "0                             0          1054   \n",
       "0                             3          1015   \n",
       "0                             0           888   \n",
       "0                             0          1180   \n",
       "0                             0           823   \n",
       "0                             0           193   \n",
       "0                             0          1400   \n",
       "0                             0           942   \n",
       "0                             0           954   \n",
       "\n",
       "  Predicted CDS with InterProScan match Total InterProScan matches  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     1                          1  \n",
       "0                                     0                          0  \n",
       "0                                     3                          6  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "0                                     0                          0  \n",
       "\n",
       "[22 rows x 30 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analyses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get list of available analysis files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "analysis_ids = analyses['id'].values\n",
    "analyses_file_info = {}\n",
    "analysis_files = check_available_sample_analysis_data(api_root, analysis_ids)\n",
    "analyses_file_info[study_id]=analysis_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subselect SSU tsv files and download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "tsv_files = get_SSU_tsv_file_list_for_study(analyses_file_info, study_id)\n",
    "file_root = f\"../metagenomics/tmp/{study_id}/\"\n",
    "\n",
    "get_tsv_files(tsv_files, file_root)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_root = \"../metagenomics/tmp/MGYS00002998/\"\n",
    "otu_df, count_df = build_otu_count_df_and_lookup_df(file_root)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "study_id = \"MGYS00002998\"\n",
    "file_root = f\"../metagenomics/tmp/{study_id}\"\n",
    "sample_id = \"MGYA00212970\"\n",
    "file_name = f\"{sample_id}__FASTQ_SSU_OTU.tsv\"\n",
    "\n",
    "file_path = os.path.join(file_root, file_name)\n",
    "read_otu_tsv_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"21\" halign=\"left\">Count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_ID</th>\n",
       "      <th>199</th>\n",
       "      <th>3611</th>\n",
       "      <th>3616</th>\n",
       "      <th>3689</th>\n",
       "      <th>4370</th>\n",
       "      <th>4511</th>\n",
       "      <th>4843</th>\n",
       "      <th>5198</th>\n",
       "      <th>5378</th>\n",
       "      <th>5592</th>\n",
       "      <th>...</th>\n",
       "      <th>262985</th>\n",
       "      <th>263057</th>\n",
       "      <th>263087</th>\n",
       "      <th>263481</th>\n",
       "      <th>263494</th>\n",
       "      <th>263805</th>\n",
       "      <th>264712</th>\n",
       "      <th>264947</th>\n",
       "      <th>265138</th>\n",
       "      <th>265348</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MGYA00212970</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>460.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212971</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212972</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212973</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1302.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212974</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1167.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212975</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>713.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212976</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212977</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212978</th>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>466.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212979</th>\n",
       "      <td>26.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>285.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212980</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212981</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212982</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212983</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>656.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212984</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212985</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>283.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212987</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>109.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212988</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>760.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212989</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212990</th>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGYA00212991</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>974.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22 rows × 759 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count                                                          \\\n",
       "OTU_ID       199    3611   3616   3689   4370   4511   4843   5198   5378     \n",
       "Sample_id                                                                     \n",
       "MGYA00212970    NaN    NaN    NaN   12.0    NaN    NaN    7.0    NaN    1.0   \n",
       "MGYA00212971    NaN    NaN    NaN    7.0    1.0    NaN    NaN    NaN    NaN   \n",
       "MGYA00212972    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212973    NaN    NaN    NaN    5.0    1.0    NaN    5.0    1.0    2.0   \n",
       "MGYA00212974    1.0    NaN    NaN    8.0    4.0    NaN    1.0    NaN    1.0   \n",
       "MGYA00212975    NaN    NaN    NaN    8.0    1.0    1.0    3.0    NaN    2.0   \n",
       "MGYA00212976    NaN    NaN    NaN    NaN    NaN    NaN    1.0    NaN    NaN   \n",
       "MGYA00212977    2.0    NaN    NaN    1.0    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212978   18.0    NaN    NaN    NaN    NaN    NaN    7.0    NaN    NaN   \n",
       "MGYA00212979   26.0    NaN    NaN    NaN    1.0    NaN    NaN    NaN    3.0   \n",
       "MGYA00212980    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212981    NaN    NaN    NaN    NaN    1.0    NaN    9.0    NaN    NaN   \n",
       "MGYA00212982    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212983    NaN    1.0    NaN    5.0    1.0    NaN    4.0    NaN    3.0   \n",
       "MGYA00212984    1.0    NaN    NaN    NaN    NaN    NaN    1.0    1.0    NaN   \n",
       "MGYA00212985    4.0    NaN    NaN    1.0    NaN    NaN    2.0    NaN    2.0   \n",
       "MGYA00212986    NaN    1.0    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212987    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   \n",
       "MGYA00212988    NaN    NaN    2.0    8.0    NaN    NaN    4.0    NaN    NaN   \n",
       "MGYA00212989    NaN    NaN    NaN    NaN    4.0    NaN    NaN    NaN    NaN   \n",
       "MGYA00212990    3.0    2.0    NaN    NaN    3.0    NaN    NaN    NaN    NaN   \n",
       "MGYA00212991    NaN    NaN    NaN    4.0    1.0    NaN    4.0    NaN    NaN   \n",
       "\n",
       "                     ...                                                    \\\n",
       "OTU_ID       5592    ... 262985 263057 263087  263481 263494 263805 264712   \n",
       "Sample_id            ...                                                     \n",
       "MGYA00212970    NaN  ...    2.0    NaN    NaN   460.0    1.0    NaN    NaN   \n",
       "MGYA00212971    NaN  ...    1.0    NaN    2.0  1000.0    2.0    NaN    2.0   \n",
       "MGYA00212972    NaN  ...    NaN    NaN    NaN     6.0    NaN    NaN    NaN   \n",
       "MGYA00212973    NaN  ...    NaN    NaN    NaN  1302.0    1.0    NaN    NaN   \n",
       "MGYA00212974    1.0  ...    1.0    NaN    1.0  1167.0    NaN    NaN    6.0   \n",
       "MGYA00212975    NaN  ...    3.0    NaN    NaN   713.0    NaN    NaN    3.0   \n",
       "MGYA00212976    NaN  ...    NaN    NaN    NaN     NaN    NaN    1.0    NaN   \n",
       "MGYA00212977    NaN  ...    1.0    NaN    NaN    31.0    NaN    NaN    NaN   \n",
       "MGYA00212978    NaN  ...    7.0    NaN    NaN   466.0    NaN    NaN    NaN   \n",
       "MGYA00212979    NaN  ...    NaN    NaN    NaN   285.0    NaN    NaN    NaN   \n",
       "MGYA00212980    NaN  ...    NaN    NaN   10.0     8.0    NaN    NaN    NaN   \n",
       "MGYA00212981    NaN  ...    NaN    NaN    1.0    43.0    NaN    NaN    NaN   \n",
       "MGYA00212982    NaN  ...    NaN    NaN    2.0     NaN    NaN    NaN    NaN   \n",
       "MGYA00212983    NaN  ...    1.0    NaN    NaN   656.0    NaN    NaN    NaN   \n",
       "MGYA00212984    1.0  ...    1.0    NaN    NaN    34.0    NaN    NaN    NaN   \n",
       "MGYA00212985    NaN  ...    NaN    NaN    4.0    45.0    3.0    NaN    1.0   \n",
       "MGYA00212986    NaN  ...    NaN    NaN    NaN   283.0    NaN    NaN    NaN   \n",
       "MGYA00212987    NaN  ...    NaN    NaN    NaN   109.0    NaN    NaN    NaN   \n",
       "MGYA00212988    NaN  ...    2.0    2.0    1.0   760.0    1.0    NaN   10.0   \n",
       "MGYA00212989    NaN  ...    NaN    NaN    NaN     NaN    NaN    NaN    NaN   \n",
       "MGYA00212990    NaN  ...    NaN    NaN    NaN     NaN    NaN    NaN    3.0   \n",
       "MGYA00212991    1.0  ...    1.0    NaN    1.0   974.0    1.0    NaN    1.0   \n",
       "\n",
       "                                   \n",
       "OTU_ID       264947 265138 265348  \n",
       "Sample_id                          \n",
       "MGYA00212970    NaN    NaN    NaN  \n",
       "MGYA00212971    NaN    NaN    NaN  \n",
       "MGYA00212972    NaN    NaN    NaN  \n",
       "MGYA00212973    NaN    NaN    1.0  \n",
       "MGYA00212974    NaN    NaN    3.0  \n",
       "MGYA00212975    NaN    NaN    2.0  \n",
       "MGYA00212976    NaN    NaN    NaN  \n",
       "MGYA00212977    NaN    NaN    NaN  \n",
       "MGYA00212978    NaN    NaN    NaN  \n",
       "MGYA00212979    NaN    NaN    NaN  \n",
       "MGYA00212980    NaN    NaN    NaN  \n",
       "MGYA00212981    NaN    NaN    NaN  \n",
       "MGYA00212982    NaN    6.0    NaN  \n",
       "MGYA00212983    2.0    NaN    1.0  \n",
       "MGYA00212984    NaN    NaN    NaN  \n",
       "MGYA00212985    NaN    NaN    NaN  \n",
       "MGYA00212986    NaN    NaN    NaN  \n",
       "MGYA00212987    NaN    NaN    NaN  \n",
       "MGYA00212988    NaN    NaN    NaN  \n",
       "MGYA00212989    NaN    NaN    NaN  \n",
       "MGYA00212990    NaN    NaN    NaN  \n",
       "MGYA00212991    NaN    NaN    NaN  \n",
       "\n",
       "[22 rows x 759 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_df\n",
    "sample_by_otu_df = count_df.pivot_table(columns=[\"OTU_ID\"], index=[\"Sample_id\"], values=[\"Count\"])\n",
    "sample_by_otu_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "dlopen(/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so, 0x0002): tried: '/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (no such file), '/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[1;32m/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/scripts/tutorial_test.ipynb Cell 18\u001b[0m line \u001b[0;36m1\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/scripts/tutorial_test.ipynb#X62sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mdiversity\u001b[39;00m \u001b[39mimport\u001b[39;00m alpha_diversity\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/scripts/tutorial_test.ipynb#X62sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m sample_by_otu_df\u001b[39m.\u001b[39mcolumns\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/__init__.py:11\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39m# Copyright (c) 2013--, scikit-bio development team.\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[39m#\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m      9\u001b[0m \n\u001b[1;32m     10\u001b[0m \u001b[39m# Add skbio.io to sys.modules to prevent cycles in our imports\u001b[39;00m\n\u001b[0;32m---> 11\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mio\u001b[39;00m  \u001b[39m# noqa\u001b[39;00m\n\u001b[1;32m     12\u001b[0m \u001b[39m# imports included for convenience\u001b[39;00m\n\u001b[1;32m     13\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39msequence\u001b[39;00m \u001b[39mimport\u001b[39;00m Sequence, DNA, RNA, Protein, GeneticCode\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/io/__init__.py:244\u001b[0m\n\u001b[1;32m    242\u001b[0m import_module(\u001b[39m'\u001b[39m\u001b[39mskbio.io.format.blast6\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m    243\u001b[0m import_module(\u001b[39m'\u001b[39m\u001b[39mskbio.io.format.blast7\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m--> 244\u001b[0m import_module(\u001b[39m'\u001b[39;49m\u001b[39mskbio.io.format.clustal\u001b[39;49m\u001b[39m'\u001b[39;49m)\n\u001b[1;32m    245\u001b[0m import_module(\u001b[39m'\u001b[39m\u001b[39mskbio.io.format.embl\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m    246\u001b[0m import_module(\u001b[39m'\u001b[39m\u001b[39mskbio.io.format.fasta\u001b[39m\u001b[39m'\u001b[39m)\n",
      "File \u001b[0;32m/opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py:126\u001b[0m, in \u001b[0;36mimport_module\u001b[0;34m(name, package)\u001b[0m\n\u001b[1;32m    124\u001b[0m             \u001b[39mbreak\u001b[39;00m\n\u001b[1;32m    125\u001b[0m         level \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m \u001b[39m1\u001b[39m\n\u001b[0;32m--> 126\u001b[0m \u001b[39mreturn\u001b[39;00m _bootstrap\u001b[39m.\u001b[39;49m_gcd_import(name[level:], package, level)\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/io/format/clustal.py:148\u001b[0m\n\u001b[1;32m    139\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[1;32m    140\u001b[0m \u001b[39m# Copyright (c) 2013--, scikit-bio development team.\u001b[39;00m\n\u001b[1;32m    141\u001b[0m \u001b[39m#\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    144\u001b[0m \u001b[39m# The full license is in the file LICENSE.txt, distributed with this software.\u001b[39;00m\n\u001b[1;32m    145\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[1;32m    147\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mio\u001b[39;00m \u001b[39mimport\u001b[39;00m create_format, ClustalFormatError\n\u001b[0;32m--> 148\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39malignment\u001b[39;00m \u001b[39mimport\u001b[39;00m TabularMSA\n\u001b[1;32m    151\u001b[0m clustal \u001b[39m=\u001b[39m create_format(\u001b[39m'\u001b[39m\u001b[39mclustal\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m    154\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_label_line_parser\u001b[39m(record):\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/alignment/__init__.py:203\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39mr\u001b[39m\u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39mAlignments (:mod:`skbio.alignment`)\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[39m===================================\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    192\u001b[0m \n\u001b[1;32m    193\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m    195\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[1;32m    196\u001b[0m \u001b[39m# Copyright (c) 2013--, scikit-bio development team.\u001b[39;00m\n\u001b[1;32m    197\u001b[0m \u001b[39m#\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    200\u001b[0m \u001b[39m# The full license is in the file LICENSE.txt, distributed with this software.\u001b[39;00m\n\u001b[1;32m    201\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[0;32m--> 203\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39m_tabular_msa\u001b[39;00m \u001b[39mimport\u001b[39;00m TabularMSA\n\u001b[1;32m    204\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39m_pairwise\u001b[39;00m \u001b[39mimport\u001b[39;00m (\n\u001b[1;32m    205\u001b[0m     local_pairwise_align_nucleotide, local_pairwise_align_protein,\n\u001b[1;32m    206\u001b[0m     local_pairwise_align, global_pairwise_align_nucleotide,\n\u001b[1;32m    207\u001b[0m     global_pairwise_align_protein, global_pairwise_align,\n\u001b[1;32m    208\u001b[0m     make_identity_substitution_matrix, local_pairwise_align_ssw\n\u001b[1;32m    209\u001b[0m )\n\u001b[1;32m    210\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39malignment\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39m_ssw_wrapper\u001b[39;00m \u001b[39mimport\u001b[39;00m (\n\u001b[1;32m    211\u001b[0m     StripedSmithWaterman, AlignmentStructure)\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/alignment/_tabular_msa.py:17\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mscipy\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mstats\u001b[39;00m\n\u001b[1;32m     16\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39m_base\u001b[39;00m \u001b[39mimport\u001b[39;00m SkbioObject\n\u001b[0;32m---> 17\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mmetadata\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39m_mixin\u001b[39;00m \u001b[39mimport\u001b[39;00m MetadataMixin, PositionalMetadataMixin\n\u001b[1;32m     18\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39msequence\u001b[39;00m \u001b[39mimport\u001b[39;00m Sequence\n\u001b[1;32m     19\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39msequence\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39m_grammared_sequence\u001b[39;00m \u001b[39mimport\u001b[39;00m GrammaredSequence\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/__init__.py:27\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39mr\u001b[39m\u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39mMetadata (:mod:`skbio.metadata`)\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[39m================================\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[39m   IntervalMetadata\u001b[39;00m\n\u001b[1;32m     17\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m     19\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[1;32m     20\u001b[0m \u001b[39m# Copyright (c) 2013--, scikit-bio development team.\u001b[39;00m\n\u001b[1;32m     21\u001b[0m \u001b[39m#\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[39m# The full license is in the file LICENSE.txt, distributed with this software.\u001b[39;00m\n\u001b[1;32m     25\u001b[0m \u001b[39m# ----------------------------------------------------------------------------\u001b[39;00m\n\u001b[0;32m---> 27\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39m_interval\u001b[39;00m \u001b[39mimport\u001b[39;00m Interval, IntervalMetadata\n\u001b[1;32m     29\u001b[0m __all__ \u001b[39m=\u001b[39m [\u001b[39m'\u001b[39m\u001b[39mInterval\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mIntervalMetadata\u001b[39m\u001b[39m'\u001b[39m]\n",
      "File \u001b[0;32m~/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_interval.py:13\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mcopy\u001b[39;00m\n\u001b[1;32m     11\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mfunctools\u001b[39;00m\n\u001b[0;32m---> 13\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39m.\u001b[39;00m\u001b[39m_intersection\u001b[39;00m \u001b[39mimport\u001b[39;00m IntervalTree\n\u001b[1;32m     14\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mskbio\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mutil\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39m_decorator\u001b[39;00m \u001b[39mimport\u001b[39;00m experimental, classonlymethod\n\u001b[1;32m     17\u001b[0m \u001b[39mclass\u001b[39;00m \u001b[39mInterval\u001b[39;00m:\n",
      "\u001b[0;31mImportError\u001b[0m: dlopen(/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so, 0x0002): tried: '/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (no such file), '/Users/madisond/Desktop/classes/2023/fall/bi270/metagenomics/.intel_env/lib/python3.11/site-packages/skbio/metadata/_intersection.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))"
     ]
    }
   ],
   "source": [
    "from skbio.diversity import alpha_diversity\n",
    "sample_by_otu_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Constructed from biom file\n",
      "#OTU ID\tSSU_rRNA\n",
      "209171\t1.2960754834361553e-05\n",
      "104349\t1.2960754834361553e-05\n",
      "5820\t0.1403390533464669\n",
      "143852\t3.888226450308466e-05\n",
      "132227\t1.2960754834361553e-05\n",
      "167520\t1.2960754834361553e-05\n",
      "196447\t2.5921509668723107e-05\n",
      "23596\t0.00015552905801233865\n",
      "126258\t0.02051687490279434\n",
      "41540\t1.2960754834361553e-05\n",
      "161694\t7.776452900616932e-05\n",
      "105940\t0.00016848981284670018\n",
      "195826\t0.0005184301933744621\n",
      "230815\t0.0008165275545647778\n",
      "70828\t0.024081082482243766\n",
      "73517\t2.5921509668723107e-05\n",
      "131038\t2.5921509668723107e-05\n",
      "236210\t2.5921509668723107e-05\n",
      "191935\t1.2960754834361553e-05\n",
      "197365\t1.2960754834361553e-05\n",
      "11441\t1.2960754834361553e-05\n",
      "74723\t7.776452900616932e-05\n",
      "87232\t7.776452900616932e-05\n",
      "197178\t1.2960754834361553e-05\n",
      "97006\t2.5921509668723107e-05\n",
      "214817\t1.2960754834361553e-05\n",
      "108524\t3.888226450308466e-05\n",
      "49655\t1.2960754834361553e-05\n",
      "110343\t2.5921509668723107e-05\n",
      "133023\t2.5921509668723107e-05\n",
      "174745\t1.2960754834361553e-05\n",
      "79734\t1.2960754834361553e-05\n",
      "127789\t1.2960754834361553e-05\n",
      "173872\t1.2960754834361553e-05\n",
      "176861\t0.00015552905801233865\n",
      "166388\t1.2960754834361553e-05\n",
      "122311\t0.00023329358701850797\n",
      "74418\t7.776452900616932e-05\n",
      "32535\t1.2960754834361553e-05\n",
      "10060\t0.0023588573798538027\n",
      "224072\t0.0005573124578775468\n",
      "123930\t1.2960754834361553e-05\n",
      "123406\t5.1843019337446214e-05\n",
      "20503\t2.5921509668723107e-05\n",
      "127984\t2.5921509668723107e-05\n",
      "220272\t2.5921509668723107e-05\n",
      "84827\t3.888226450308466e-05\n",
      "171261\t2.5921509668723107e-05\n",
      "142749\t0.01143138576390689\n",
      "235232\t2.5921509668723107e-05\n",
      "104076\t6.480377417180777e-05\n",
      "29440\t0.00025921509668723106\n",
      "187742\t1.2960754834361553e-05\n",
      "46651\t0.0025403079475348643\n",
      "23045\t0.028759914977448287\n",
      "205225\t1.2960754834361553e-05\n",
      "33269\t0.00018145056768106174\n",
      "118827\t0.00016848981284670018\n",
      "165834\t7.776452900616932e-05\n",
      "41432\t5.1843019337446214e-05\n",
      "93912\t1.2960754834361553e-05\n",
      "39384\t0.00023329358701850797\n",
      "39922\t1.2960754834361553e-05\n",
      "45001\t0.0014645652962828556\n",
      "201803\t1.2960754834361553e-05\n",
      "168693\t6.480377417180777e-05\n",
      "5370\t1.2960754834361553e-05\n",
      "177912\t0.0002851366063559542\n",
      "171391\t1.2960754834361553e-05\n",
      "152234\t1.2960754834361553e-05\n",
      "137900\t0.0001944113225154233\n",
      "54635\t6.480377417180777e-05\n",
      "31015\t2.5921509668723107e-05\n",
      "149740\t1.2960754834361553e-05\n",
      "17600\t1.2960754834361553e-05\n",
      "195412\t2.5921509668723107e-05\n",
      "71145\t2.5921509668723107e-05\n",
      "160662\t1.2960754834361553e-05\n",
      "152476\t7.776452900616932e-05\n",
      "123387\t6.480377417180777e-05\n",
      "175458\t1.2960754834361553e-05\n",
      "175998\t7.776452900616932e-05\n",
      "28808\t0.0001425683031779771\n",
      "234282\t1.2960754834361553e-05\n",
      "145648\t3.888226450308466e-05\n",
      "126805\t3.888226450308466e-05\n",
      "58292\t0.01302555860853336\n",
      "101463\t0.0069728861008865155\n",
      "129308\t0.0005702732127119084\n",
      "20337\t1.2960754834361553e-05\n",
      "111351\t1.2960754834361553e-05\n",
      "109759\t1.2960754834361553e-05\n",
      "233822\t1.2960754834361553e-05\n",
      "174348\t0.004225206076001867\n",
      "31237\t2.5921509668723107e-05\n",
      "135303\t1.2960754834361553e-05\n",
      "134190\t0.024158847011249936\n",
      "93849\t0.02936907045466328\n",
      "145560\t0.0004795479288713775\n",
      "7656\t0.0005443517030431852\n",
      "57471\t1.2960754834361553e-05\n",
      "15245\t2.5921509668723107e-05\n",
      "184240\t0.03888226450308466\n",
      "49542\t2.5921509668723107e-05\n",
      "62836\t1.2960754834361553e-05\n",
      "196428\t3.888226450308466e-05\n",
      "162330\t2.5921509668723107e-05\n",
      "46186\t3.888226450308466e-05\n",
      "93008\t1.2960754834361553e-05\n",
      "52683\t1.2960754834361553e-05\n",
      "13082\t3.888226450308466e-05\n",
      "58934\t1.2960754834361553e-05\n",
      "99474\t0.0007387630255586085\n",
      "38831\t1.2960754834361553e-05\n",
      "215130\t2.5921509668723107e-05\n",
      "81496\t3.888226450308466e-05\n",
      "138266\t5.1843019337446214e-05\n",
      "164009\t1.2960754834361553e-05\n",
      "120834\t2.5921509668723107e-05\n",
      "60150\t2.5921509668723107e-05\n",
      "59442\t9.072528384053087e-05\n",
      "2315\t0.006648867230027476\n",
      "115126\t0.0003240188708590388\n",
      "53265\t0.0007906060448960548\n",
      "184560\t1.2960754834361553e-05\n",
      "201601\t1.2960754834361553e-05\n",
      "136491\t1.2960754834361553e-05\n",
      "167366\t0.0006998807610555239\n",
      "69752\t0.0001944113225154233\n",
      "184349\t0.00010368603867489243\n",
      "44081\t1.2960754834361553e-05\n",
      "110462\t0.00033697962569340036\n",
      "71736\t0.00010368603867489243\n",
      "120891\t3.888226450308466e-05\n",
      "140615\t0.0027606407797190107\n",
      "9760\t0.10972575042770491\n",
      "120351\t0.00024625434185286953\n",
      "158717\t0.0001425683031779771\n",
      "234932\t1.2960754834361553e-05\n",
      "33966\t2.5921509668723107e-05\n",
      "95237\t1.2960754834361553e-05\n",
      "119884\t2.5921509668723107e-05\n",
      "129441\t0.000609155477214993\n",
      "85171\t3.888226450308466e-05\n",
      "112303\t0.0001425683031779771\n",
      "237507\t1.2960754834361553e-05\n",
      "200331\t0.21542070610192338\n",
      "40397\t1.2960754834361553e-05\n",
      "76398\t0.00011664679350925399\n",
      "141625\t0.0015812120897921095\n",
      "99667\t5.1843019337446214e-05\n",
      "214424\t1.2960754834361553e-05\n",
      "2660\t1.2960754834361553e-05\n",
      "111794\t2.5921509668723107e-05\n",
      "116471\t3.888226450308466e-05\n",
      "96747\t1.2960754834361553e-05\n",
      "198062\t0.0020737207734978485\n",
      "56838\t3.888226450308466e-05\n",
      "219226\t1.2960754834361553e-05\n",
      "8060\t6.480377417180777e-05\n",
      "133949\t3.888226450308466e-05\n",
      "234094\t5.1843019337446214e-05\n",
      "94344\t0.0002203328321841464\n",
      "89749\t1.2960754834361553e-05\n",
      "85210\t0.008852195551868941\n",
      "174087\t1.2960754834361553e-05\n",
      "136388\t7.776452900616932e-05\n",
      "56423\t0.0004147441546995697\n",
      "64993\t1.2960754834361553e-05\n",
      "42572\t0.0001425683031779771\n",
      "218352\t1.2960754834361553e-05\n",
      "137076\t7.776452900616932e-05\n",
      "75093\t0.0003240188708590388\n",
      "219034\t1.2960754834361553e-05\n",
      "38127\t0.00011664679350925399\n",
      "226487\t0.0004147441546995697\n",
      "223669\t0.07991601430867333\n",
      "189865\t1.2960754834361553e-05\n",
      "12614\t1.2960754834361553e-05\n",
      "7\t3.888226450308466e-05\n",
      "105915\t0.0003888226450308466\n",
      "11614\t5.1843019337446214e-05\n",
      "57843\t1.2960754834361553e-05\n",
      "77090\t0.0008942920835709471\n",
      "141040\t1.2960754834361553e-05\n",
      "102978\t0.0015812120897921095\n",
      "191460\t3.888226450308466e-05\n",
      "135744\t6.480377417180777e-05\n",
      "34790\t0.0005832339675462699\n",
      "186187\t0.00016848981284670018\n",
      "220917\t5.1843019337446214e-05\n",
      "196225\t0.0013479185027736014\n",
      "76602\t1.2960754834361553e-05\n",
      "176527\t0.00033697962569340036\n",
      "236026\t0.0001944113225154233\n",
      "151638\t1.2960754834361553e-05\n",
      "229412\t0.031416869718492405\n",
      "150820\t0.0003888226450308466\n",
      "143314\t0.00023329358701850797\n",
      "10125\t2.5921509668723107e-05\n",
      "29779\t0.0002851366063559542\n",
      "147035\t7.776452900616932e-05\n",
      "193647\t1.2960754834361553e-05\n",
      "44047\t1.2960754834361553e-05\n",
      "110157\t9.072528384053087e-05\n",
      "135217\t0.0027347192700502877\n",
      "78144\t3.888226450308466e-05\n",
      "132341\t2.5921509668723107e-05\n",
      "209876\t5.1843019337446214e-05\n",
      "123643\t0.0005573124578775468\n",
      "25797\t2.5921509668723107e-05\n",
      "136978\t0.0008683705739022241\n",
      "76095\t0.0005184301933744621\n",
      "5646\t0.0012571932189330707\n",
      "147727\t0.00015552905801233865\n",
      "43222\t7.776452900616932e-05\n",
      "24850\t0.00015552905801233865\n",
      "197704\t1.2960754834361553e-05\n",
      "82291\t2.5921509668723107e-05\n",
      "97987\t0.0002851366063559542\n",
      "16232\t2.5921509668723107e-05\n",
      "231904\t0.010251957073979988\n",
      "97967\t0.0005054694385401005\n",
      "141708\t2.5921509668723107e-05\n",
      "94120\t0.0006221162320493546\n",
      "72418\t2.5921509668723107e-05\n",
      "33984\t1.2960754834361553e-05\n",
      "161683\t0.011742443879931567\n",
      "154436\t0.0004017833998652081\n",
      "4670\t6.480377417180777e-05\n",
      "14312\t2.5921509668723107e-05\n",
      "46150\t1.2960754834361553e-05\n",
      "23209\t1.2960754834361553e-05\n",
      "151898\t0.0011405464254238167\n",
      "225450\t2.5921509668723107e-05\n",
      "14963\t1.2960754834361553e-05\n",
      "5608\t2.5921509668723107e-05\n",
      "120558\t3.888226450308466e-05\n",
      "220424\t1.2960754834361553e-05\n",
      "48742\t1.2960754834361553e-05\n",
      "52402\t0.08535953133910519\n",
      "63291\t0.00042770490953393124\n",
      "66238\t0.008554098190678625\n",
      "25209\t2.5921509668723107e-05\n",
      "40813\t1.2960754834361553e-05\n",
      "43012\t0.0005573124578775468\n",
      "163308\t0.0007776452900616932\n",
      "66626\t0.00010368603867489243\n",
      "165703\t5.1843019337446214e-05\n",
      "140836\t0.001477526051117217\n",
      "203240\t3.888226450308466e-05\n",
      "137036\t3.888226450308466e-05\n",
      "96006\t1.2960754834361553e-05\n",
      "101540\t1.2960754834361553e-05\n",
      "39577\t2.5921509668723107e-05\n",
      "134392\t1.2960754834361553e-05\n",
      "127494\t0.00018145056768106174\n",
      "103412\t1.2960754834361553e-05\n",
      "137958\t0.00011664679350925399\n",
      "230479\t7.776452900616932e-05\n",
      "106490\t0.0005184301933744621\n",
      "164636\t0.0005573124578775468\n",
      "18086\t0.0001425683031779771\n",
      "76463\t1.2960754834361553e-05\n",
      "64056\t0.0010498211415832857\n",
      "162454\t7.776452900616932e-05\n",
      "87328\t6.480377417180777e-05\n",
      "35470\t0.00020737207734978485\n",
      "133665\t0.0006350769868837161\n",
      "70441\t0.0001425683031779771\n",
      "137401\t1.2960754834361553e-05\n",
      "180653\t1.2960754834361553e-05\n",
      "161905\t0.000725802270724247\n",
      "114920\t0.0015682513349577478\n",
      "14613\t0.0015423298252890249\n",
      "44134\t0.002151485302504018\n",
      "52617\t2.5921509668723107e-05\n",
      "138146\t0.00011664679350925399\n",
      "178952\t0.001853387941313702\n",
      "130904\t2.5921509668723107e-05\n",
      "99611\t5.1843019337446214e-05\n",
      "219187\t2.5921509668723107e-05\n",
      "60687\t2.5921509668723107e-05\n",
      "90601\t2.5921509668723107e-05\n",
      "170329\t9.072528384053087e-05\n",
      "26019\t9.072528384053087e-05\n",
      "44993\t0.0002851366063559542\n",
      "50891\t0.0006869200062211623\n",
      "43744\t1.2960754834361553e-05\n",
      "219365\t6.480377417180777e-05\n",
      "38824\t1.2960754834361553e-05\n",
      "36204\t2.5921509668723107e-05\n",
      "64323\t2.5921509668723107e-05\n",
      "173153\t0.0004147441546995697\n",
      "85536\t2.5921509668723107e-05\n",
      "171591\t0.0005443517030431852\n",
      "134918\t0.0004925086837057391\n",
      "101644\t0.0007128415158898855\n",
      "80341\t5.1843019337446214e-05\n",
      "199134\t0.0006609984965524392\n",
      "33908\t0.0003240188708590388\n",
      "57344\t1.2960754834361553e-05\n",
      "210936\t2.5921509668723107e-05\n",
      "1268\t0.00011664679350925399\n",
      "221500\t5.1843019337446214e-05\n",
      "171652\t1.2960754834361553e-05\n",
      "93001\t6.480377417180777e-05\n",
      "1161\t5.1843019337446214e-05\n",
      "165588\t0.003888226450308466\n",
      "120573\t1.2960754834361553e-05\n",
      "65967\t0.0005443517030431852\n",
      "186210\t0.0015812120897921095\n",
      "142899\t0.0002851366063559542\n",
      "144277\t1.2960754834361553e-05\n",
      "38947\t3.888226450308466e-05\n",
      "147726\t0.00015552905801233865\n",
      "85728\t0.0009720566125771165\n",
      "214434\t0.008891077816372026\n",
      "113943\t2.5921509668723107e-05\n",
      "112760\t0.0003110581160246773\n",
      "152509\t0.00033697962569340036\n",
      "112224\t3.888226450308466e-05\n",
      "208866\t0.0002851366063559542\n",
      "43358\t2.5921509668723107e-05\n",
      "227844\t0.0018922702058167867\n",
      "108867\t1.2960754834361553e-05\n",
      "72557\t6.480377417180777e-05\n",
      "186424\t7.776452900616932e-05\n",
      "5100\t0.00010368603867489243\n",
      "231658\t1.2960754834361553e-05\n",
      "231590\t0.00011664679350925399\n",
      "25886\t2.5921509668723107e-05\n",
      "233120\t0.0001944113225154233\n",
      "16049\t0.0006998807610555239\n",
      "21739\t0.0002203328321841464\n",
      "23761\t2.5921509668723107e-05\n",
      "94591\t0.0005702732127119084\n",
      "2614\t1.2960754834361553e-05\n",
      "197485\t0.00010368603867489243\n",
      "119872\t0.0008813313287365857\n",
      "228570\t7.776452900616932e-05\n",
      "234358\t3.888226450308466e-05\n",
      "79162\t6.480377417180777e-05\n",
      "83155\t0.002034838508994764\n",
      "237651\t0.0002980973611903157\n",
      "141355\t1.2960754834361553e-05\n",
      "104088\t2.5921509668723107e-05\n",
      "184743\t1.2960754834361553e-05\n",
      "144119\t0.00015552905801233865\n",
      "100053\t1.2960754834361553e-05\n",
      "85114\t3.888226450308466e-05\n",
      "89078\t3.888226450308466e-05\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from biom import load_table, parse_table, Table\n",
    "import biom\n",
    "# biom.__version__\n",
    "table = load_table(file_path)\n",
    "# # print(table.shape())\n",
    "# df = table.to_dataframe()\n",
    "table\n",
    "# print(table.ids(axis=\"observation\"))\n",
    "# table\n",
    "normed = table.norm(axis='sample', inplace=False)\n",
    "print(normed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># OTU ID</th>\n",
       "      <th>ERR204745</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>Taxa Dict</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>103181</td>\n",
       "      <td>356.0</td>\n",
       "      <td>sk__Bacteria</td>\n",
       "      <td>{'Super Kingdom': 'Bacteria', 'Kingdom': None,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>38816</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sk__Bacteria;k__;p__;c__;o__;f__;g__;s__marine...</td>\n",
       "      <td>{'Super Kingdom': 'Bacteria', 'Kingdom': None,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>83222</td>\n",
       "      <td>3.0</td>\n",
       "      <td>sk__Bacteria;k__;p__Acidobacteria</td>\n",
       "      <td>{'Super Kingdom': 'Bacteria', 'Kingdom': None,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>127050</td>\n",
       "      <td>9.0</td>\n",
       "      <td>sk__Bacteria;k__;p__Actinobacteria;c__Acidimic...</td>\n",
       "      <td>{'Super Kingdom': 'Bacteria', 'Kingdom': None,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>162670</td>\n",
       "      <td>11.0</td>\n",
       "      <td>sk__Bacteria;k__;p__Actinobacteria;c__Acidimic...</td>\n",
       "      <td>{'Super Kingdom': 'Bacteria', 'Kingdom': None,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>139256</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...</td>\n",
       "      <td>{'Super Kingdom': 'Chloroplast', 'Kingdom': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>93004</td>\n",
       "      <td>73.0</td>\n",
       "      <td>sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...</td>\n",
       "      <td>{'Super Kingdom': 'Chloroplast', 'Kingdom': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>70257</td>\n",
       "      <td>12.0</td>\n",
       "      <td>sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...</td>\n",
       "      <td>{'Super Kingdom': 'Chloroplast', 'Kingdom': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>233398</td>\n",
       "      <td>59.0</td>\n",
       "      <td>sk__Mitochondria</td>\n",
       "      <td>{'Super Kingdom': 'Mitochondria', 'Kingdom': N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>88503</td>\n",
       "      <td>4.0</td>\n",
       "      <td>sk__Mitochondria;k__;p__;c__;o__;f__;g__;s__un...</td>\n",
       "      <td>{'Super Kingdom': 'Mitochondria', 'Kingdom': N...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>315 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     # OTU ID  ERR204745                                           taxonomy  \\\n",
       "0      103181      356.0                                       sk__Bacteria   \n",
       "1       38816        1.0  sk__Bacteria;k__;p__;c__;o__;f__;g__;s__marine...   \n",
       "2       83222        3.0                  sk__Bacteria;k__;p__Acidobacteria   \n",
       "3      127050        9.0  sk__Bacteria;k__;p__Actinobacteria;c__Acidimic...   \n",
       "4      162670       11.0  sk__Bacteria;k__;p__Actinobacteria;c__Acidimic...   \n",
       "..        ...        ...                                                ...   \n",
       "310    139256        1.0  sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...   \n",
       "311     93004       73.0  sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...   \n",
       "312     70257       12.0  sk__Chloroplast;k__;p__;c__;o__;f__;g__;s__unc...   \n",
       "313    233398       59.0                                   sk__Mitochondria   \n",
       "314     88503        4.0  sk__Mitochondria;k__;p__;c__;o__;f__;g__;s__un...   \n",
       "\n",
       "                                             Taxa Dict  \n",
       "0    {'Super Kingdom': 'Bacteria', 'Kingdom': None,...  \n",
       "1    {'Super Kingdom': 'Bacteria', 'Kingdom': None,...  \n",
       "2    {'Super Kingdom': 'Bacteria', 'Kingdom': None,...  \n",
       "3    {'Super Kingdom': 'Bacteria', 'Kingdom': None,...  \n",
       "4    {'Super Kingdom': 'Bacteria', 'Kingdom': None,...  \n",
       "..                                                 ...  \n",
       "310  {'Super Kingdom': 'Chloroplast', 'Kingdom': No...  \n",
       "311  {'Super Kingdom': 'Chloroplast', 'Kingdom': No...  \n",
       "312  {'Super Kingdom': 'Chloroplast', 'Kingdom': No...  \n",
       "313  {'Super Kingdom': 'Mitochondria', 'Kingdom': N...  \n",
       "314  {'Super Kingdom': 'Mitochondria', 'Kingdom': N...  \n",
       "\n",
       "[315 rows x 4 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      356.0\n",
       "1        1.0\n",
       "2        3.0\n",
       "3        9.0\n",
       "4       11.0\n",
       "       ...  \n",
       "299    101.0\n",
       "300      2.0\n",
       "301      3.0\n",
       "302      2.0\n",
       "303     14.0\n",
       "Name: ERR204745, Length: 304, dtype: float64"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_table[csv_table['taxonomy'].str.startswith(\"sk__Bacteria\")]['ERR204745']\n",
    "# csv_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# kingdom_df = pd.DataFrame()\n",
    "# phylum_df = pd.DataFrame()\n",
    "# class_df = pd.DataFrame()\n",
    "# order_df = pd.DataFrame()\n",
    "# family_df = pd.DataFrame()\n",
    "# genus_df = pd.DataFrame()\n",
    "# species_df = pd.DataFrame()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
